meilisearch-lib/src/index_controller/mod.rs

```diff
@@ -356,7 +358,18 @@ where
     }
 
     pub async fn register_update(&self, uid: String, update: Update) -> Result<Task> {
-        let index_uid = IndexUid::new(uid)?;
+        // cleaner attempts that did not compile:
+        //      let index_uid = IndexUid::from_str(&uid)?;
+        //      let index_uid: std::result::Result<_, IndexResolverError> = IndexUid::from_str(&uid).into()?;
+        // bad implementation but compiles
+        let index_uid = match IndexUid::from_str(&uid) {
+            Ok(uid) => uid,
+            Err(e) => {
+                let e: IndexResolverError = e.into();
+                Err(e)?
+            }
+        };
```

Comment:
Not sure of the reason why it is struggling to infer the error type but this Should do the trick:

```suggestion
let index_uid = IndexUid::from_str(&uid).map_err(IndexResolverError::from)?;
```

Reply:
It looks like the reason it is struggling to infer the conversion function (i.e. `Into::into`) is because multiple ones can match:

```
note: multiple `impl`s satisfying `IndexControllerError: ~const std::convert::From<_>` found
   --> meilisearch-lib/src/index_controller/error.rs:18:17
    |
18  | #[derive(Debug, thiserror::Error)]
    |                 ^^^^^^^^^^^^^^^^
...
40  | internal_error!(IndexControllerError: JoinError, UpdateFileStoreError);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
41  |
42  | impl From<actix_web::error::PayloadError> for IndexControllerError {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: and another `impl` found in the `core` crate: `impl<T> std::convert::From<T> for T;`
    = note: required because of the requirements on the impl of `FromResidual<std::result::Result<Infallible, _>>` for `std::result::Result<tasks::task::Task, IndexControllerError>`
```

Reply:
Ah that makes sense! Thank you for the explanation!